### PR TITLE
fix(aws-logs): issue 4492, Using LogGroup.fromLogGroupArn creates log…

### PR DESCRIPTION
**Pull request by jordanst3wart**
issue 4492, Using LogGroup.fromLogGroupArn creates logGroup name with invalid characters.

The logGroup ARN looks like:
```
arn:aws:logs:us-west-1:123456789012:log-group:/mystack-testgroup-12ABC1AB12A1:*
```
ref: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-loggroup.html
Were the loggroup name is `/mystack-testgroup-12ABC1AB12A1`, currently the name would be returned as `/mystack-testgroup-12ABC1AB12A1:*`, which is wrong.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jordanst3wart** [comment]
Fixes: https://github.com/aws/aws-cdk/issues/4492

**Comment by rix0rrr** [comment]
You are correct, it IS wrong, but this is CFN behaviour and works like this for correctly granting permissions to log streams (iirc).

What issues is this causing for yoU?

**Comment by rix0rrr** [comment]
Oh got it.

File: packages/@aws-cdk/aws-logs/lib/log-group.ts [hunk 1]
```diff
@@ -310,7 +310,17 @@ export class LogGroup extends LogGroupBase {
   public static fromLogGroupArn(scope: Construct, id: string, logGroupArn: string): ILogGroup {
     class Import extends LogGroupBase {
       public readonly logGroupArn = logGroupArn;
-      public readonly logGroupName = Stack.of(scope).parseArn(logGroupArn, ':').resourceName!;
+      public readonly logGroupName = this.getLogGroupName();
+
+      private getLogGroupName() {
+          const resourceName = Stack.of(scope).parseArn(logGroupArn, ':').resourceName!;
+          const components = resourceName.split(':');
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-logs/lib/log-group.ts]: Unfortunately, you can't do this (always). The value of `resourceName` might be a token (and look like `"${Token[LogGroup.Ref.1234]}"`).

At the very least, add a check that says `if (!Token.isUnresolved(resourceName))` around it.

In fact, in case `logGroupArn` *is* a Token, because of the way `{ Fn::Split }` works this would already work, in which case we should just return `resourceName` directly.

File: packages/@aws-cdk/aws-logs/test/test.loggroup.ts [hunk 2]
```diff
@@ -115,7 +115,7 @@ export = {
     const stack2 = new Stack();
 
     // WHEN
-    const imported = LogGroup.fromLogGroupArn(stack2, 'lg', 'arn:aws:logs:us-east-1:123456789012:log-group:my-log-group');
+    const imported = LogGroup.fromLogGroupArn(stack2, 'lg', 'arn:aws:logs:us-east-1:123456789012:log-group:my-log-group:*');
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-logs/test/test.loggroup.ts]: Please don't modify existing tests, just add a new one. 

Additionally, this test is now asserting something that is not entirely right. The `:*` at the end is not part of the ARN of the log group. I am okay with optimistically recognizing this situation and Doing The Right Thing, but passing a proper LogGroup ARN should not all of a sudden become illegal.

**Comment by jordanst3wart** [comment]
> I would like to note that your description of your PR is different than what the change appears to be achieving.
> 
> Your PR states that the "returned name" is wrong because CFN returns a ":*".
> 
> Your change achieves that a user must now import a LogGroup given a literal string with a ":_" in it, at which point the "_" will be stripped off.
> 
> What is your actual issue/what are you trying to achieve?

tldr: I can't recreate the original error now. I'm looking to close the PR until I can re-create the error.

I was getting an error when importing a logGroup arn without the`:*` with the `LogGroup.fromLogGroupArn` method. When I changed the logGroup to have a `:*`, the error seemed to be resolved, but another error occurred with the logGroup name being wrong.

Now, I cannot recreate the original import error. It may be due to updating from `1.12` to `1.13`. So I'm closing this issue, if I come across this issue again. I will re-open it.
